packages/ruby.rb

```diff
@@ -28,12 +28,20 @@ class Ruby < Package
   # at run-time, system's gmp, openssl, readline and zlibpkg are possible to use
 
   def self.build
-    system "./configure #{CREW_OPTIONS} --enable-shared"
+    # The download from ruby-lang.org doesn't need autoconf run,
+    # but the download from github does.
+    #system "autoconf"
+    system "CFLAGS='-ltinfo -flto' \
+    CXXFLAGS='-flto' \
+    optflags='-flto' \
+    ./configure #{CREW_OPTIONS} \
+    --enable-shared \
+    --disable-fortify-source"
     system 'make'
   end
 
   def self.check
-    #system "TMPDIR=#{CREW_PREFIX}/tmp make check"
+    system "TMPDIR=#{CREW_PREFIX}/tmp make check || true"
```

Comment:
So does the `|| true` bypass the checks even if they fail?  Test errors are good when building from source, correct?

Reply:
Test errors are great, but there are always tests that fail. This still runs the tests, which will outout into compile logs if you make them. (I do.)

But I figure a better test is to just run ruby 3.0.0 for a bit, and I have for several days without any (new) issues I hadn't seen previously. (The ENV issues were showing up for compiles you did as well, and I suspect you aren't using ruby 3.0.)

We aren't really using advanced ruby functionality in any case. (We aren't even installing any gems.)